jumpscale/sals/zos/container_stats.py

```diff
@@ -0,0 +1,101 @@
+from urllib.parse import urlparse
+import requests
+import json
+import redis
+import sys
+
+class ContainerStatsMonitor:
+    def __init__(self):
+        self.host = None
+        self.port = None
+        self.channel = None
+
+        self.redis = None
+        self.pubsub = None
+        self.previous = None
+
+    def endpoint(self, endpoint):
+        params = urlparse(endpoint)
+
+        self.host = params.netloc.split(':')[0]
+        self.port = params.netloc.split(':')[1]
+        self.channel = params.path[1:]
+
+        return self.connect()
+
+    #
+    # Reservation Parser
+    #
+    def reservation(self, url):
+        reservation = requests.get(url).json()
+
+        if len(reservation['stats']) == 0:
+            raise RuntimeError("No statistics endpoint defined")
+
+        if reservation['stats'][0]['type'] != "redis":
+            raise RuntimeError("Invalid statistics type, not supported")
+
+        return self.endpoint(reservation['stats'][0]['data']['endpoint'])
+
+    def connect(self):
+        print("[+] connecting: [%s:%s / %s]" % (self.host, self.port, self.channel))
```

Comment:
I think we could use logging here instead of `print`

Reply:
print is used elsewhere too :D

Reply:
do you have example about logging ?

Reply:
Sure:

https://github.com/threefoldtech/js-sdk/blob/cd4c954a24304a2e8edbbe54d5d9409f570f6d2e/jumpscale/clients/explorer/explorer.py#L25-L27

Other methods with more options are available too:
![Uploading Screenshot from 2020-10-09 17-47-09.png…]()